prisma/interview.prisma

```diff
@@ -1,53 +1,61 @@
-enum InterviewType {
-  PHONE_SCREEN
-  TECHNICAL
-  BEHAVIORAL
-  ONSITE
-  FINAL
-}
-
-enum InterviewStatus {
-  SCHEDULED
-  COMPLETED
-  CANCELLED
-  RESCHEDULED
-}
-
-enum InterviewLocationType {
-  PHYSICAL
-  VIRTUAL
-}
-
 model Interview {
+  // Fields
   id                    String                @id @default(uuid())
-  jobApplicationId      String
-  jobApplication        JobApplication        @relation(fields: [jobApplicationId], references: [id])
   type                  InterviewType
   scheduledDate         DateTime
   durationMinutes       Int
   interviewLocationType InterviewLocationType
-  intervieweeId         String
-  interviewee           User                  @relation("InterviewInterviewee", fields: [intervieweeId], references: [id])
-  interviewers          User[]                @relation("InterviewInterviewers") //  (array of User IDs of interviewers)
-  companyContactId      String?
-  companyContact        CompanyContact?       @relation(fields: [companyContactId], references: [id])
   preparationNotes      String?
   status                InterviewStatus
   feedback              String?
   candidateFeedback     String?
   nextSteps             String?
+
+  // Custodial Fields
   createdAt             DateTime              @default(now())
   updatedAt             DateTime              @updatedAt
   deletedAt             DateTime?
 
+  // Custodial Relations
   createdById String
   createdBy   User   @relation("InterviewCreatedBy", fields: [createdById], references: [id])
   updatedById String
   updatedBy   User   @relation("InterviewUpdatedBy", fields: [updatedById], references: [id])
 
+  // Profile Relations
+  intervieweeId         String
+  interviewee           Profile               @relation("InterviewInterviewee", fields: [intervieweeId], references: [id])
+  interviewers          Profile[]             @relation("InterviewInterviewers")
+
+  // Relationships
+  jobApplicationId      String
+  jobApplication        JobApplication        @relation(fields: [jobApplicationId], references: [id])
+  companyContactId      String?
+  companyContact        CompanyContact?       @relation(fields: [companyContactId], references: [id])
+
   @@index([jobApplicationId])
```

Comment:
Do we need indexing at this point of time? That should be introduced when we know which query is being used often